process/smartContract/hooks/blockChainHook.go

```diff
@@ -287,14 +303,22 @@ func (bh *BlockChainHookImpl) LastTimeStamp() uint64 {
 
 // LastRandomSeed returns the random seed from the last committed block
 func (bh *BlockChainHookImpl) LastRandomSeed() []byte {
+	if bh.flagDisableBlockchainOperations.IsSet() {
+		return make([]byte, 0)
+	}
+
 	if !check.IfNil(bh.blockChain.GetCurrentBlockHeader()) {
 		return bh.blockChain.GetCurrentBlockHeader().GetRandSeed()
 	}
-	return []byte{}
+	return make([]byte, 0)
 }
 
 // LastEpoch returns the epoch from the last committed block
 func (bh *BlockChainHookImpl) LastEpoch() uint32 {
+	if bh.flagDisableBlockchainOperations.IsSet() {
```

Comment:
the only one which needs disable is the GetBlockhash(nonce uint64)

Reply:
done